### DISCUSSION TITLE
Pad Operator and Space: Is it time to have a prints filter operator?

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#6192] [Q&A] [answered]
In https://tiddlywiki.com/#pad%20Operator
It states you can use a padding-string so

```
<$list filter="[range[1,15]pad[2],[0]]">
<$text text=<<currentTiddler>> /><br>
</$list>
```
works fine! but how if I want to pad with space!

```
<$list filter="[range[1,15]pad[2],[ ]]">
<$text text=<<currentTiddler>> /><br>
</$list>
```
This doesnot work!

I use this time to ask is it possible to have a  `prints`, stands for print string, operator to be able to format a filter output?
This will eliminate the requirement of many concatenations!

Side note: we can get idea from python simple format function here!

**Comment by ericshulman** [comment, marked as answer]
Actually, `[range[1,15]pad[2],[ ]]` **does** work.  It's just that when the **output** is displayed, the whitespace is being collapsed by the browser's HTML rendering engine.

You can see the whitespace in the output by wrapping the code within `<pre>...</pre>`, like this:
`<pre><$list filter="[range[1,15]pad[2],[ ]]"><$text text=<<currentTiddler>> /><br></$list></pre>`

**Reply by kookma** [reply]
Thank you Eric! Yes, pad seems to work!
